### PR TITLE
fix(datagrid): miss ellipsis on the large header

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
we have regression with [Talend/ui#2022](https://github.com/Talend/ui/pull/2022). chrome don't handle flex on button element very well. The overflow hidden is not respected.

**What is the chosen solution to this problem?**
So, we remove button by div and restore accessibility action(press space and enter) to select the column

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/datagrid/src/components/DefaultHeaderRenderer/DefaultHeaderRenderer.component.js [hunk 1]
```diff
@@ -14,16 +15,23 @@ export const HEADER_RENDERER_COMPONENT = 'headerGrid';
...
-			<button
-				className={classNames(theme['td-header'], 'td-header')}
-				onClick={onHeaderClick}
-				onKeyDown={onHeaderKeyDown}
-				type="button"
-			>
+		<div
+			className={classNames(theme['td-header-component'], 'td-header-component')}
+			onClick={onHeaderClick}
+			onKeyDown={onHeaderKeyDown}
```
Comment (frassinier) [inline packages/datagrid/src/components/DefaultHeaderRenderer/DefaultHeaderRenderer.component.js]: They suggest onKeyPress instead
> https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md#no-static-element-interactions

**Comment by jare-talend** [comment]
> Flex in buttons work on chrome, but not on firefox.
> But on firefox, adding a span in the button that will display its children using flex works.
> 
> Please don't recode a button with a div, just use button and fix style using html/css.

you can try on this : https://codepen.io/jare-talend/pen/vPOOer

This works on Firefox and not on chrome

File: packages/datagrid/src/components/DefaultHeaderRenderer/DefaultHeaderRenderer.test.js [hunk 2]
```diff
@@ -34,8 +35,10 @@ describe('#DefaultHeaderGrid', () => {
 				displayName="Title"
 			/>,
 		);
-		wrapper.find('button').simulate('click');
-		wrapper.find('button').simulate('keydown', event);
+		wrapper.find('div[role="button"]').simulate('click');
+		wrapper.find('div[role="button"]').simulate('keydown', event);
+		wrapper.find('div[role="button"]').simulate('keydown', { keyCode: keycode('space') });
+		wrapper.find('div[role="button"]').simulate('keydown', { keyCode: keycode('enter') });
```
[the remark this answers is not in the document]
Comment (jare-talend) [reply packages/datagrid/src/components/DefaultHeaderRenderer/DefaultHeaderRenderer.test.js]: I test accessibility is good on this element

[the remark this answers is not in the document]
Comment (frassinier) [reply packages/datagrid/src/components/DefaultHeaderRenderer/DefaultHeaderRenderer.test.js]: Not really since you can't expect anything after :)

**Comment by jare-talend** [comment]
I retry with span. this seems works :)

**Pull request merged**